crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -367,6 +373,18 @@ impl<'w, T: Component> DerefMut for ResMut<'w, T> {
     }
 }
 
+impl<'w, T: Component> AsRef<T> for ResMut<'w, T> {
+    fn as_ref(&self) -> &T {
+        &*self
+    }
+}
+
+impl<'w, T: Component> AsMut<T> for ResMut<'w, T> {
+    fn as_mut(&mut self) -> &mut T {
+        &mut *self
```

Comment:
```suggestion
        self.ticks.set_changed(self.change_tick);
        &mut *self
```
I think we make need to set the changed status in case any mutation occurs with the `&mut T`

Reply:
Do we need to do that for `Mut` also?